CDCgov/trusted-intermediary
# Ingest JSON Functionality

This PR adds JSON functionality to the application. The application is able to ingest JSON format and returns an order receipt acknowledgement in JSON.
### What Was Added
- Order.java
- OrderMessage.java
- pojo unit tests
- e2e tests
- Updated the e2e client to not throw an exception when a 4xx or 5xx status code was returned
- logging (might need more or less)
- parseBody helper function (Destined to change)
- Formatter interface
- JacksonFormatter.java
- FormatterProcessingException.java
- Changed the injection framework to work when we only registered an implementation under an interface
- Fixed a problem when running the e2e tests made the application start failing

### How to Test
1. compile code: `./gradlew clean build` (you might need to have app running first: `./gradlew :app:run`)
2. All tests must pass
3. Make sure app is running: `./gradlew :app:shadowJar` then `docker compose up --build`
4. send POST request via curl or Postman (`curl -L -X POST 'http://localhost:8080/v1/etor/order' -H 'Content-Type: application/json' --data-raw '{
	"id": "an ID",
	"destination": "Massachusetts",
	"createdAt": "2022-12-21T08:34:27Z",
	"client": "MassGeneral"
}'`)
5.  check that you get your destination, client, and body in the return message.


## Issue

This PR deals with:
- story/76/Ingest an "order" like JSON object for the request
- story/76/Output some kind of acceptance JSON object for the response
